packages/remark-lint-no-undefined-references/index.js

```diff
@@ -305,12 +346,26 @@ const remarkLintNoUndefinedReferences = lintRule(
         if (
           !generated({position: pos}) &&
           !(normalizeIdentifier(id) in map) &&
-          !allow.has(normalizeIdentifier(id))
+          !isAllowed(id)
         ) {
           file.message('Found reference to undefined definition', pos)
         }
       }
     }
+
+    /**
+     * @param {string} id
+     * @returns {boolean}
+     */
+    function isAllowed(id) {
+      if (strings.size > 0 && strings.has(normalizeIdentifier(id))) {
+        return true
+      }
+      if (regexes.some((regex) => regex.test(id))) {
+        return true
+      }
+      return false
```

Comment:
Everything can be replaced with:

```suggestion
      const normalized = normalizeIdentifier(id)
      return strings.has(normalized) || regexes.some((regex) => regex.test(normalized))
```

*   I think `normalized` should be used for the regex too.
*   And I don’t think the `set.size` check is needed?

Reply:
> * I think normalized should be used for the regex too.

Hm, I had thought about it but then concluded that coming up with the right regex would require internal knowledge about the `normalizeIdentifier(...)` function whereas for the string case it doesn't because both the string and the ID are normalized. WDYT?

> * And I don’t think the `set.size` check is needed?

I added this check to avoid an unnecessay call to `normalizeIdentifier(...)` in case the `strings` set is empty. If we conclude that the regex tests should receive normalized IDs, then you're right. If the regex tests should receive _unnormalized_ IDs, the above could still be "simplified" to

```js
return (
  (strings.size > 0 && strings.has(normalizeIdentifier(id))) ||
  regexes.some((regex) => regex.test(id))
)
```

although I find the formatting produced by Prettier less readable TBH. But it's your call in the end.

Reply:
responded below in a whole review